src/Pretzel.Logic/Configuration.cs

```diff
@@ -55,7 +59,8 @@ private void EnsureDefaults()
             _defaultsConfiguration = new DefaultsConfiguration(_config);
         }
 
-        internal void ReadFromFile()
+        [OnImportsSatisfied]
+        public void ReadFromFile()
```

Comment:
I'd prefer to avoid making this public, is it possible for OnImportSatisfied to work on internal or private methods?

Reply:
After a quick check and the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.composition.onimportssatisfiedattribute?view=dotnet-plat-ext-2.1#remarks) it must be public or internal.